editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/globalconfiguration/beans/wizards/pages/GlobalBeanWizardPage.java

```diff
@@ -0,0 +1,205 @@
+		argsPropsGroup.setLayout(GridLayoutFactory.swtDefaults().numColumns(4).create());
+		argsPropsGroup.setLayoutData(GridDataFactory.fillDefaults().grab(true, true).span(4, 1).create());
+
+		beanArgsTable = new ArgumentStyleChildTableControl(argsPropsGroup, SWT.NULL);
+		beanArgsTable.setLayoutData(GridDataFactory.fillDefaults().grab(false, false).span(4, 2).create());
+		beanArgsTable.setInput(this.parent);
+
+		Group beanPropsGroup = new Group(composite, SWT.NONE);
+		beanPropsGroup.setText("Bean Properties");
+		beanPropsGroup.setLayout(GridLayoutFactory.swtDefaults().numColumns(4).create());
+		beanPropsGroup.setLayoutData(GridDataFactory.fillDefaults().grab(true, true).span(4, 1).create());
+
+		beanPropsTable = new PropertyStyleChildTableControl(beanPropsGroup, SWT.NULL);
+		beanPropsTable.setLayoutData(GridDataFactory.fillDefaults().grab(false, false).span(4, 2).create());
+		beanPropsTable.setInput(this.parent);
+
+		setControl(composite);
+		WizardPageSupport.create(this, dbc);
+	}
+
+	/**
+	 * @param composite
+	 */
+	private void createIdLine(Composite composite) {
+		Label idLabel = new Label(composite, SWT.NONE);
+		idLabel.setText(UIMessages.GlobalEndpointWizardPage_idFieldLabel);
+		Text idText = new Text(composite, SWT.BORDER);
+		idText.setLayoutData(GridDataFactory.fillDefaults().indent(10, 0).grab(true, false).span(3, 1).create());
+		UpdateValueStrategy strategy = new UpdateValueStrategy();
+		strategy.setBeforeSetValidator(new NewBeanIdValidator(parent));
+
+		final IObservableValue<?> idObservable = PojoProperties.value(GlobalBeanWizardPage.class, "id", String.class).observe(this); //$NON-NLS-1$
+		Binding binding = dbc.bindValue(WidgetProperties.text(SWT.Modify).observe(idText), idObservable, strategy, null);
+		ControlDecorationSupport.create(binding, SWT.LEFT | SWT.TOP);
+	}
+
+	/**
+	 * @param composite
+	 */
+	private void createClassLine(Composite composite) {
+		Label classLabel = new Label(composite, SWT.NONE);
+		classLabel.setText("Class");
+		Text classText = new Text(composite, SWT.BORDER);
+		classText.setLayoutData(GridDataFactory.fillDefaults().indent(10, 0).grab(true, false).create());
+		UpdateValueStrategy strategy = new UpdateValueStrategy();
+		strategy.setBeforeSetValidator(new BeanClassExistsValidator());
+
+		classObservable = PojoProperties.value(GlobalBeanWizardPage.class, "classname", String.class).observe(this); //$NON-NLS-1$
+		Binding binding = dbc.bindValue(WidgetProperties.text(SWT.Modify).observe(classText), classObservable, strategy, null);
+		ControlDecorationSupport.create(binding, SWT.LEFT | SWT.TOP);
+	}
+
+	private void createBrowseButton(Composite composite) {
+		Button browseBeanButton = new Button(composite, SWT.PUSH);
+		browseBeanButton.setText("...");
+		browseBeanButton.addSelectionListener(new SelectionAdapter() {
+			@Override
+			public void widgetSelected(SelectionEvent event) {
+				final IProject project = CamelUtils.project();
```

Comment:
better to avoid to use the CamelUtils.project() as it is linking this code the workbench state.
On AbstractCamelElement we can getCamelFile().getResource().getProject()